common/web/gesture-recognizer/src/tools/unit-test-resources/src/touchpathTurtle.ts

```diff
@@ -1,130 +1,119 @@
+     *      `move`, returning Subsegmentation by wrapping THIS class's
+     *      version of the methods & tracking 'start' and 'end' state
+     *      CumulativePathStats as before.
+     *
+     * https://github.com/keymanapp/keyman/pull/7440/ should hopefully
+     * maintain enough relevant preserved history to bypass the need for
+     * `git blame`.
+     */
+  }
+
+  wait(totalTime: number, repeatInterval: number) {
     if(repeatInterval < 0 || totalTime < 0) {
       throw new Error("Invalid parameter value:  may not be negative!");
     }
 
-    const segmentBaseStats = this.currentChop;
-    const startSample = this.currentSample;
+    const startSample = this.location;
 
     // Base sample always exists in advance.
-    let currentStats = this.currentStats;
     for(let timeDelta = 0; timeDelta < totalTime; timeDelta += repeatInterval) {
       let sample = {...startSample};
       sample.t += timeDelta;
-
-      currentStats = currentStats.extend(sample);
-    }
-
-    this.currentSample = {...startSample};
-    this.currentSample.t += totalTime;
-    this.currentChop = currentStats;
-    this.currentStats = currentStats.extend(this.currentSample);
-
-    const pathComponent = {
-      stats: this.currentStats.deaccumulate(segmentBaseStats),
-      baseAccumulation: segmentBaseStats,
-      endingAccumulation: this.currentStats
+      this.trackSample(sample);
     }
 
-    this._pathSegments.push(pathComponent);
-    return pathComponent;
+    let currentSample = {...startSample};
+    currentSample.t += totalTime;
+    this.trackSample(currentSample);
   }
 
-  move(angleInDegrees: number, distance: number, time: number, interval: number): Subsegmentation {
+  move(angleInDegrees: number, distance: number, time: number, interval: number) {
     if(distance < 0 || time < 0 || interval < 0) {
       // negative angles are actually fine, so we don't filter those.
       throw new Error("Invalid parameter value:  may not be negative!");
     }
 
     // Angle is measured clockwise from the DOM's <0, -1>.
-    let angle = angleInDegrees * Math.PI / 180;
+    let angle = (90 - angleInDegrees) * Math.PI / 180;
     let xDistance = Math.cos(angle) * distance;
     let yDistance = -Math.sin(angle) * distance;
 
     const xTickDist = xDistance * interval / time;
     const yTickDist = yDistance * interval / time;
 
-    const segmentBaseStats = this.currentChop;
-    const startSample = this.currentSample;
+    const startSample = this.location;
 
     // While _definitely_ not a perfect match for a real input recording... we'll just interpolate evenly.
     let currentSample = {...startSample};
-    let currentStats  = this.currentStats;
 
     for(let timeDelta = 0; timeDelta < time; timeDelta += interval) {
       currentSample = {
         targetX: currentSample.targetX + xTickDist,
         targetY: currentSample.targetY + yTickDist,
-        t: currentSample.t + timeDelta
+        t: startSample.t + timeDelta
```

Comment:
Pulled from #7440 - the time value wasn't being set properly for constructed samples.